product_of_all_other_numbers/product_of_all_other_numbers.py

```diff
@@ -2,10 +2,11 @@
 Input: a List of integers
 Returns: a List of integers
 '''
+from functools import reduce
 def product_of_all_other_numbers(arr):
-    # Your code here
-
-    pass
+    product = reduce((lambda x, y: x * y), arr)
+    result = map(lambda z: product / z, arr)
+    return list(result)
```

Comment:
🤯 🤯 🤯 🤯 🤯 🤯 

This is a really clever solution. There is a way to achieve this with an `O(1) space complexity. It will require a different approach, but see if you can find it for tonight's objective.